builtin/providers/gitlab/resource_gitlab_project.go

```diff
@@ -56,6 +60,10 @@ func resourceGitlabProject() *schema.Resource {
 				Default:      "private",
 			},
 
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
```

Comment:
This isn't needed.  The reason you cite for adding it is to help work with the `gitlab_project_hook` resource, but if you look at the acceptance tests for that code you'll see that the id is already exposed and used in its tests:  https://github.com/hashicorp/terraform/blob/master/builtin/providers/gitlab/resource_gitlab_project_hook_test.go#L189   Its just an automatically exported attribute that all resources get.